app/user.go

```diff
@@ -872,12 +871,20 @@ func (a *App) SetProfileImageFromFile(userId string, file io.Reader) *model.AppE
 	buf := new(bytes.Buffer)
 	err = png.Encode(buf, img)
 	if err != nil {
-		return model.NewAppError("SetProfileImage", "api.user.upload_profile_user.encode.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, model.NewAppError("SetProfileImage", "api.user.upload_profile_user.encode.app_error", nil, err.Error(), http.StatusInternalServerError)
 	}
+	return buf.Bytes(), nil
+}
 
+func (a *App) SetProfileImageFromFile(userId string, file io.Reader) *model.AppError {
+
+	buf, err := a.AdjustImage(file)
+	if err != nil {
+		return err
+	}
 	path := "users/" + userId + "/profile.png"
 
-	if _, err := a.WriteFile(buf, path); err != nil {
+	if _, err := a.WriteFile(bytes.NewReader(buf), path); err != nil {
```

Comment:
```suggestion
	if _, err := a.WriteFile(buf, path); err != nil {
```

I think with the above suggested change to `(*App).AdjustImage` you can use the `Buffer` directly without having to create a new reader.